CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+  - Run `make test-integration` locally to ensure all tests pass in CI.
+  - Note: since `make test-integration` is time-expensive, before you can run your test to ensure it passes with: 
+```
+poetry run pytest src/tests/integration -k 'test_foundry_xml_report' --maxfail=1 --verbose --durations=0 --numprocesses=4 --dist=worksteal
+```
+
+### Run kontrol with a custom `pyk` / `kevm`
+- If your fix involves changes in `pyk` or `kevm`, you might want to run Kontrol with a custom `pyk`/`kevm`. If that is the case, follow these [instructions](https://github.com/runtimeverification/kontrol/issues/319).
+
+### Adding support for a new cheatcode
+The foundry success predicate as well as the addresses needed to define cheatcodes are in [foundry.md](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/foundry.md).  The rules to support cheatcodes are located in [cheatcodes.md](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/cheatcodes.md). At the top of the file there is the subconfiguration needed to implement the cheatcodes. The [structure of execution](https://github.com/runtimeverification/kontrol/blob/master/src/kontrol/kdist/cheatcodes.md#structure-of-execution) presents how calls to the Foundry cheatcode address are handled. After understanding this infrastructure, implementing a new cheatcode resumes to:
+##### 1. Add a new `call_foundry` rule for the cheatcode
+```k
+ rule [foundry.call.cheatcode]:
+         <k> #call_foundry SELECTOR ARGS => logic implementation of the cheatcode ... </k>
+      requires SELECTOR ==Int selector ( "cheatcode signature" )
+```
+
+##### 2. Add a rule for the selector
+Example:
+```k
+rule ( selector ( "assume(bool)" )                             => 1281615202 )
+```
+If you are implementing a cheatcode that belongs to the list of selectors for non-implemented cheatcodes (end of the file), please move the rule for that selector into the list for the implemented cheatcodes.
+
+##### 3. Editing the subconfiguration
+If the cheatcode implementation requires aditional information to be stored in the subconfiguration please provide documentation on the new cell added. 
+
+##### 4. Add tests for the cheatcode into CI
+- Check in the current [test-suite](https://github.com/runtimeverification/kontrol/tree/master/src/tests/integration/test-data/foundry/test) if there are any tests for the cheatcode being implemented. If not, please add the tests you consider necessary. 
+- Make sure the tests are in `foundry-prove-all` list (or `foundry-fail` if that is the case)
+- Make sure the tests are removed from `foundry-prove-skip`
+- Run the tests locally with `kontrol prove --no-use-booster`. If they take more than 300s then they should be in `foundry-prove-skip-legacy`, otherwise they can be removed from that list
```

Comment:
```suggestion
- Run the tests locally with `kontrol prove --no-use-booster`. If a test takes more than 300s, it should be added to `foundry-prove-skip-legacy`, otherwise it can be removed from that list
```

Reply:
If the test is already in `foundry-prove-skip-legacy`, I don't think it would be executed by `kontrol prove --no-use-booster`.

Reply:
Good catch! I edited this point to make that clear.